From d64710d16859d2587e7b3426b77d43081c51767f Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 14 Feb 2023 06:33:17 +0200
Subject: [PATCH] [SW-123505] gaudi2: boot_info: remove fake security

This patch removes usage of obsolete HL_BOOT_INFO_FAKE_SECURITY.

In relevant function modified to use cold rst flop instead.

Change-Id: I140f32969a4d24c1136d1d299ae0105dda82e05f
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 72cca40..0535c68 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -369,7 +369,6 @@
 	u32 data_size, rst_src;
 	u32 nic_en_local;
 	u32 fw_upd_is_dis = 0;
-	u32 fake_security = 0;
 
 	if (!boot_info_get(HL_BOOT_INFO_FIT_BOOT_ERROR, &boot_err,
 			   sizeof(boot_err), &data_size)) {
@@ -424,18 +423,6 @@
 		 * current reset sequence.
 		 */
 		fw_info.reset_cause = HL_FW_RESET_CAUSE_UNKNOWN;
-	}
-
-	/*
-	 * Preserve fake security boot info if exists. This info will be later utilized by u-boot,
-	 * zephyr, and agent to check whether fake security is enabled or not
-	 */
-	if (!boot_info_get(HL_BOOT_INFO_FAKE_SECURITY, &fake_security,
-				sizeof(fake_security), &data_size)) {
-		preboot_puts_hex("Preserving fake security boot info:", fake_security, 0);
-		if (boot_info_set(HL_BOOT_INFO_FAKE_SECURITY, &fake_security,
-					sizeof(fake_security)))
-			preboot_puts("Failed to preserve fake security boot info\n");
 	}
 }
 
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 4806b0b..793fbd2 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -145,15 +145,11 @@
 
 bool preboot_fake_security_is_sig_validation_enabled(void)
 {
-	u32 fake_security;
-	u32 copy_sz;
-	bool sig_validation_en = false;
+	struct gaudi2_cold_rst_data cold_rst_data;
 
-	if (!boot_info_get(HL_BOOT_INFO_FAKE_SECURITY, &fake_security, sizeof(fake_security),
-				&copy_sz))
-		sig_validation_en = !!(fake_security & FAKE_SECURITY_SIG_VALIDATION_ENABLE);
+	cold_rst_data.data = regmap_readl(mmCOLD_RST_DATA);
 
-	return sig_validation_en;
+	return cold_rst_data.fake_sig_validation_en;
 }
 
 bool preboot_is_image_sig_supported(void)
